dask/dataframe/dask_expr/_collection.py

```diff
@@ -749,12 +749,21 @@ def isin(self, values):
             except TypeError:
                 pass
             if not any(is_dask_collection(v) for v in values):
-                try:
-                    values = np.fromiter(values, dtype=object)
-                except ValueError:
-                    # Numpy 1.23 supports creating arrays of iterables, while lower
-                    # version 1.21.x and 1.22.x do not
-                    pass
+                # Avoid always passing a numpy array of object dtype
+                inferred_type = pd.api.types.infer_dtype(values, skipna=False)
+                object_like = {
+                    "mixed-integer",
+                    "decimal",
+                    "categorical",
+                    "time",
+                    "period",
+                    "mixed",
+                    "unknown-array",
+                }
```

Comment:
Is there a utility function in `pandas` that we could use here instead of this explicit list of object-like dtypes? Maybe `pandas.api.types.is_object_dtype`?

Reply:
Unfortunately not. `is_object_dtype` specifically checks for `object` or `np.dtype(np.object_)`, but it is a good suggestion. I'll open a pandas issue to see what the other pandas core devs think about it

Reply:
Got it, thanks for the extra context